exercises/bhuvan_bezawada/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,257 @@
+/**
+* PROGRAM DESCRIPTION:
+*
+* A command line program to sort shapes and display the area and perimeter 
+* of each shape.
+*
+**/
+
+// Imports
+#include <algorithm>
+#include <iostream>
+#include <math.h>
+#include <string.h>
+#include <string>
+#include <vector>
+
+// Namespace being used
+using namespace std;
+
+/**
+ *  Base class
+ **/
+class Shape {
+
+// Instance variables
+private:
+  string shapeType;
+  unsigned int numSides;
+
+// Methods
+public:
+  Shape(const string &type, int sides) {
+    shapeType = type;
+    numSides = sides;
+  }
+
+  // Methods to be overridden
+  virtual double perimeter() = 0;
+  virtual double area() = 0;
+
+  // Methods for info
+  void printType() const {
+    cout << "Shape type is: " << getShapeType() << endl;
+  }
+
+  virtual void printNumSides() const {
+    cout << "Number of sides is: " << getNumSides() << endl;
+  }
+
+  // Getters
+  string getShapeType() const { return shapeType; }
+
+  int getNumSides() const { return numSides; }
+
+  // Destructor
+  virtual ~Shape() {}
+};
+
+/**
+ *  Square class
+ **/
+class Square : public Shape {
```

Comment:
Isn't a square a special case of a rectangle? Did you consider having Square inherit from Rectangle instead?